halo/app/monitor.go

```diff
@@ -202,30 +211,32 @@ func dirSize(path string) (int64, error) {
 	return size, nil
 }
 
-// startMonitoringAPI registers metrics and health endpoints.
-// Return the function shutting down the HTTP server.
+// startMonitoringAPI starts the monitoring API serving
+// metrics and health endpoints. It returns HTTP server shutdown function.
+// Note this replaces CometBFT's prometheus server, adding a `/ready` endpoint.
 func startMonitoringAPI(
 	cfg *cmtcfg.Config,
 	asyncAbort chan<- error,
-	status *readinessStatus) func(context.Context) error {
+	status *readinessStatus,
+) func(context.Context) error {
 	mux := http.NewServeMux()
 
-	// We align both with industry standards (endpoint /metrics)
-	// and with cometBFT (endpoint /).
-	mux.Handle("/metrics", promhttp.Handler())
-	mux.Handle("/", promhttp.Handler())
+	promHandler := promhttp.HandlerFor(
+		prometheus.DefaultGatherer,
+		promhttp.HandlerOpts{MaxRequestsInFlight: cfg.Instrumentation.MaxOpenConnections},
+	)
+	mux.Handle("/metrics", promHandler) // Industry standard `/metrics` path
+	mux.Handle("/", promHandler)        // CometBFT standard `/` path
 
-	// On the `/ready` endpoint, we serve the readiness of the halo node.
-	// Additionally, in case when the node is not ready, the response status code is set to 503.
+	// Serve readiness status json at `/ready`, returning 503 if not ready.
 	mux.HandleFunc("/ready", func(w http.ResponseWriter, _ *http.Request) {
 		w.Header().Set("Content-Type", "application/json")
 
-		if !status.ready() {
-			w.WriteHeader(http.StatusServiceUnavailable)
-		}
-		if err := status.serialize(w); err != nil {
```

Comment:
there is actualy a race condition here. We read ready and then we serialiaze. Technically, the status could change in between these two steps resulting in inconsistent output. This isn't critical code, but I thought I would clean it up in any case